File: php/src/Model/WorkflowRunStep.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The state of one step within a run.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowRunStep
{
    public function __construct(
        public readonly string $stepId,
        public readonly string $name,
        public readonly string $taskType,
        public readonly string $state,
        public readonly ?string $taskId,
        public readonly ?string $workerId,
        public readonly ?int $completedAt,
        /** @var array<string, mixed>|null */
        public readonly ?array $result,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            stepId: (string) ($data['stepId'] ?? ''),
            name: (string) ($data['name'] ?? ''),
            taskType: (string) ($data['taskType'] ?? ''),
            state: (string) ($data['state'] ?? ''),
            taskId: isset($data['taskId']) ? (string) $data['taskId'] : null,
            workerId: isset($data['workerId']) ? (string) $data['workerId'] : null,
            completedAt: isset($data['completedAt']) ? (int) $data['completedAt'] : null,
            result: $data['result'] ?? null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkflowRunStep.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * The state of one step within a run.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowRunStep {
    private String stepId;
    private String name;
    private String taskType;
    private String state;
    private String taskId;
    private String workerId;
    private Long completedAt;
    private Map<String, Object> result;

    public String getStepId() { return stepId; }
    public String getName() { return name; }
    public String getTaskType() { return taskType; }
    public String getState() { return state; }
    public String getTaskId() { return taskId; }
    public String getWorkerId() { return workerId; }
    public Long getCompletedAt() { return completedAt; }
    public Map<String, Object> getResult() { return result; }
}
